### PR TITLE
fix: adjust GA to track all registry pages

### PR DESCRIPTION
**Pull request by wgwz**
## Description

Closes: https://github.com/regen-network/regen-registry/issues/1069

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/lib/ga.ts [hunk 1]
```diff
@@ -54,21 +56,32 @@ function isCookiesEnabled(): boolean {
...
-    if (
-      Cookies.get(
-        COOKIE_GATSBY_PLUGIN_GOOGLE_ANALYTICS_GDPR_COOKIES_ENABLED,
-      ) === undefined
-    ) {
-      Cookies.set(
-        COOKIE_GATSBY_PLUGIN_GOOGLE_ANALYTICS_GDPR_COOKIES_ENABLED,
-        'false',
-      );
+export function useGoogleAnalyticsInit(): void {
```
Comment (wgwz) [inline web-registry/src/lib/ga.ts:59]: the adjustment to this function is the important change so far

File: web-registry/src/lib/ga.ts [hunk 2]
```diff
@@ -54,21 +56,32 @@ function isCookiesEnabled(): boolean {
...
+  const location = useLocation();
+  const [initialized, setInitialized] = useState(false);
+
+  useEffect(() => {
+    if (process.env.NODE_ENV === 'production') {
```
Comment (wgwz) [inline web-registry/src/lib/ga.ts:64]: one thing i need help with is figuring out how to test this. i think at present we only have google analytics running for the `master` branch (NODE_ENV=production). it's a small set of changes so far, but i would still feel better knowing it's been tested. i suppose i will need to create a temporary tag for my localhost or something... if anyone knows how i can test or have advice, please LMK.

Comment (flagrede) [reply web-registry/src/lib/ga.ts:64]: Actually, all netlify builds should be run with NODE_ENV production. However, it's possible that only the events on the production domain are saved. I think it's possible to keep all events by defining environments in GA: https://support.google.com/tagmanager/answer/6311518?hl=en

**Review by flagrede — COMMENTED** [review]
This looks good!
We should probably rebase this branch on master now that `v4` is merged.

File: web-registry/src/lib/ga.ts (hunk 2 above)
Comment (wgwz) [reply web-registry/src/lib/ga.ts:64]: It turns out that we're sending data for even the deploy previews to our google analytics setup. We'll probably want to change that before going live, i.e. we probably want to have something like prod has it's own analytics and perhaps redwood, hambach, and deploy previews all have their own separated analytics. Currently it just all getes sent to the same analytics bucket. Thanks for point that out @flagrede .

I was able to test this by going to the deploy preview and I could see that there was real-time analytics coming through while I was clicking around (even on the "trade" and "activity" tabs, which i don't think were working previously).

@erikalogie i think you could test this by going here:

- https://deploy-preview-1120--regen-registry.netlify.app/
- https://analytics.google.com/analytics/web/#/realtime/rt-overview/a119338253w247735863p229931745/

the second link should bring you to the real-time view in GA for the "Regen Network (new site)" analytics, and when you're clicking around you should be able to see your activity where you hadn't previously.

**Review by haveanicedavid — COMMENTED** [review]
nice work!

**Review by flagrede — APPROVED** [review]

File: web-registry/src/lib/ga.ts (hunk 2 above)
Comment (wgwz) [reply web-registry/src/lib/ga.ts:64]: i created a new issue for making the current google analytics setup only receive production data: https://github.com/regen-network/regen-registry/issues/1133

**Review by mhagel — APPROVED** [review]

**Pull request merged**